CMSgov/mint-app
# [MINT-3029] Models approaching clearance

# MINT-3029

## Description

- Added settings and translation support for new `MODELS_APPROACHING_CLEARANCE` enum type
- Added component to home
- Added new `CalendarDate` component per USWDS style guidelines
- Added new `usePagination` hook to condense state and Truss' `Pagination` component
- Unit tests
- LD Flag

## How to test this change

1.  Select new clearance setting on settings page
2. Verify empty state
3. Add a model with a clearance start within 6 months
4. Add enough plans for pagination - verify works as expected

## PR Author Checklist




## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
